src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,205 @@
+     * @param taskList The TaskList to process CRUD operations based on user input.
+     * @return An output denoting the success / failure of an operation.
+     * @throws DukeTerminateException Shuts down the Duke by entering "bye".
+     * @throws InsufficientInputException Not enough to arguments to find/delete/done.
+     * @throws InvalidIndexException Wrong indexes from TaskList's delete and done operations.
+     * @throws InvalidTodoException Invalid format of Task entered by the user.
+     * @throws InvalidEventException Invalid format of Event entered by the user.
+     * @throws InvalidDeadlineException Invalid format of Deadline entered by the user.
+     * @throws InvalidInputException Invalid commands entered by the user.
+     */
+    public String processCommand(String command, TaskList taskList) throws DukeTerminateException,
+            InsufficientInputException, InvalidIndexException, InvalidTodoException, InvalidEventException, InvalidDeadlineException,
+            InvalidInputException {
+        String output = "";
+        if (command.equals("bye")) { // exit
+                throw new DukeTerminateException();
+            } else if (command.equals("list")) {
+                output = taskList.list();
+            } else if (command.split(" ")[0].equals("find")) { //find
+                if (command.split(" ").length < 2) {
+                    throw new InsufficientInputException();
+                }
+                String key = command.split(" ")[1].trim();
+                output = taskList.find(key);
+            } else if (command.split(" ")[0].equals("done")) { // done
+                if (command.split(" ").length < 2) {
+                    throw new InsufficientInputException();
+                }
+                int idx = Integer.parseInt(command.split(" ")[1]) - 1;
+                output = taskList.done(idx);
+            } else if (command.split(" ")[0].equals("delete")) { // delete
+                if (command.split(" ").length < 2) {
+                    throw new InsufficientInputException();
+                }
+                int idx = Integer.parseInt(command.split(" ")[1]) - 1;
+                output = taskList.delete(idx);
+            } else if (command.split(" ")[0].equals("todo")) { // add todo
+                if (command.split(" ").length < 2) {
+                    throw new InvalidTodoException();
+                }
+                command = command.substring(command.split(" ")[0].length() + 1, command.length());
+                Task task = new Task(command);
+                output = taskList.addTask(task);
+            } else if (command.split(" ")[0].equals("deadline")) { // add deadline
+                command = command.substring(command.split(" ")[0].length() + 1, command.length()).trim();
+                if (command.split("/by").length == 0) {
+                    throw new InvalidDeadlineException("OOPS! The description and time is missing. " +
```

Comment:
I noticed the exception code is repeated multiple times, would it be easier to define this in the Task class instead?